lib/pure/os.nim

```diff
@@ -3310,3 +3310,69 @@ func isValidFilename*(filename: string, maxLen = 259.Positive): bool {.since: (1
     find(f.name, invalidFilenameChars) != -1): return false
   for invalid in invalidFilenames:
     if cmpIgnoreCase(f.name, invalid) == 0: return false
+
+when defined(windows):
+  const ComputerNamePhysicalNetBIOS = 4
+  const ComputerNamePhysicalDnsHostname = 5
+
+proc getHostname*(): string {.tags: [ReadIOEffect].} =
+  ## Returns the local hostname (not the FQDN, but ComputerNamePhysicalDnsHostname)
+  ## See https://docs.microsoft.com/en-us/windows/win32/sysinfo/computer-names
+  # https://tools.ietf.org/html/rfc1035#section-2.3.1
+  # https://tools.ietf.org/html/rfc2181#section-11
+  const size = 64
```

Comment:
on my system man gethostname says:

> Host names are limited in length to {sysconf(_SC_HOST_NAME_MAX)} characters, not including > the trailing null, currently 255.

Reply:
That's why I put 256 in the first place :-) OK, I'll restore it back to 256.